src/sentry/integrations/source_code_management/repository.py

```diff
@@ -215,15 +215,15 @@ def encode_url(url: str) -> str:
                 # Preserve path separators and query params etc.
                 return urlunparse(parsed._replace(path=encoded_path))
 
-            if version:
-                scope.set_tag("stacktrace_link.tried_version", True)
-                source_url = self.check_file(repo, filepath, version)
-                if source_url:
-                    scope.set_tag("stacktrace_link.used_version", True)
-                    return encode_url(source_url)
-
-            scope.set_tag("stacktrace_link.used_version", False)
             try:
+                if version:
+                    scope.set_tag("stacktrace_link.tried_version", True)
+                    source_url = self.check_file(repo, filepath, version)
```

Comment:
Missed this check_file call before and was wondering why we were still seeing these 😭